server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -1039,66 +1039,62 @@ export function configureWebSocketServices(
 		);
 
 		// Message sent when a new signal is submitted to the router
-		socket.on(
-			"submitSignal",
-			(clientId: string, contentBatches: (IDocumentMessage | IDocumentMessage[])[]) => {
-				// Verify the user has subscription to the room.
-				const room = roomMap.get(clientId);
-				if (!room) {
+		// Elements of 'contentBatches' can be individual or nested arrays of messages
+		socket.on("submitSignal", (clientId: string, contentBatches: unknown[]) => {
```

Comment:
We can be more precise about this, right? "messages" implies a *Message type, but that is not correct, right?
We should be able to say what types we actually expect.
We can comment that type is `unknown` apart from being an array because: (1) this isn't really validated and (2) we don't need to know the type to operate on it.

Reply:
Please do still comment on what is expected.
And go ahead and say where that is sourced from. Do we have client-side gates that make us reasonably sure? In recent discussion with Teams LiveShare they were talking about more direct use of Signals. I don't know if that means they are sending through things that are not ISignalEnvelope, but would be interesting to know.

Reply:
I added some additional comments -- I'm not sure of the TLS discussion but if 'direct use of signals' is referring to what they [currently do](https://github.com/microsoft/live-share-sdk/blob/b11a930f776b2327c6e707e8074e55de680735af/packages/live-share/src/internals/ContainerSynchronizer.ts#L221) then the content they're sending still gets packaged in a signal envelope (see [here](https://github.com/microsoft/FluidFramework/blob/bae72dc359d6aa5a8ee9139337fdd9e8cd61c55b/packages/runtime/container-runtime/src/containerRuntime.ts#L2782C1-L2786C3))

Reply:
@jason-ha Signal envelopes are converted to JSON strings before being sent to the server, so I updated the expected type to be JSON string instead of ISignalEnvelope.

Reply:
I think your comments are saying that `contentBatches` is `string | string[])[]`.
So never was `ISignalEnvelope` and you didn't see that under debugger.?
To close the loop on this, where is the client code that stringifies `ISignalEnvelope` (even though we can't exactly make that type restriction in client right now)?

Reply:
Cool. So, from that point down, then we should be able to say that the content is a string or array of strings all the way until it is placed on the wire. (Clean up the internal `any`s.) But I don't think we can quite get there yet.
I tried it locally and `IDocumentDeltaConnection` is an alpha API. So, that means someone might be using its `submitSignal`. Can you do a check on Loop and Teams LiveShare for direct use of `IDocumentDeltaConnection.submitSignal`?

Reply:
It looks like Loop and TLS don't have any direct usage of `IDocumentDeltaConnection`.

Reply:
I have a draft PR that removes all of the `any`'s  #20217 -- will still need to investigate if there is any `submitSignal` usage from any layers deeper than the container runtime since some of the APIs are alpha/public.

Reply:
We probably should do a little more with the general `any`s out there. We need to get delta connection and lower to have confidence. That should be covered by #20251 which I went ahead and merged.
So, we can do better than a comment that says we checked under a debugger. We can reference the client code that emits.

Reply:
Made a change to the comment and the type to match #20251.